src/mapgen_valleys.cpp

```diff
@@ -963,23 +961,24 @@ void MapgenValleys::generateCaves(s16 max_stone_y)
 
 			// River water is not set as ground content
 			// in the default game. This can produce strange results
-			// when a cave undercuts a river. However, that's not for
+			// when a tunnel undercuts a river. However, that's not for
 			// the mapgen to correct. Fix it in lua.
 
 			if (c == CONTENT_AIR) {
-				air_above = true;
+				continue;
```

Comment:
This `continue;` statement is ineffectual.  You should remove this entire if clause if no work is being performed.